debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java

```diff
@@ -94,6 +97,14 @@ protected void start(BlockingConsumer<ChangeEvent> eventConsumer, Consumer<Throw
         }
     }
 
+    private void delaySnapshot() {
+        Duration delay = taskContext.getConfig().getSnapshotDelay();
+        if (!delay.isZero() && !delay.isNegative()) {
+            logger.info("The connector will wait for " + delay.toMillis() + " ms before proceeding");
+            DelayStrategy.constant(delay.toMillis()).sleepWhen(true);
```

Comment:
I don't think sleeping is a good idea, instead it'd be better to do it similarly to the existing `TimedBlockingReader` in the MySQL connector: keep the polling loop running, but return an empty list of records during the delay period. That way, the connector can be stopped gracefully during that time.

Reply:
So that's still my biggest concern with this change, @grzegorz8.

Reply:
Yes, I see. At first glance I don't know exactly how incorporate your idea here. I'll try to return to this issue as soon as possible.

Reply:
Hey @grzegorz8, I've pushed one commit to this PR which shows what I meant. It does the change for the new framework based connectors (SQL Server, Oracle) by delaying in `ChangeEventSourceCoordinator`. Do you think you could do it equivalently now for MongoDB and Postgres?

Reply:
I believe I can. I'll try tonight :)

Reply:
Hi @gunnarmorling. I've just added changes in MongoDB and PostgreSQL part. Please let me know if this meets your expectations.

By the way, because of your change in `ChangeEventSourceCoordinator` the connector always sleep at (re)start, provided the `snapshot.delay.ms > 0`.

Reply:
> By the way, because of your change in ChangeEventSourceCoordinator the connector always sleeps at (re)start, provided the snapshot.delay.ms > 0.

Hum, hum, that's a good point indeed. So this would only have to happen if the snapshotting source actually would do something.